loris/webapp.py

```diff
@@ -621,19 +621,18 @@ def _make_image(self, image_request, src_fp, src_format):
             (str) the fp of the new image
         '''
         # figure out paths, make dirs
-        if self.enable_caching:
-            target_fp = self.img_cache.create_dir_and_return_file_path(image_request)
-        else:
-            # random str
-            n = ''.join(random.choice(string.ascii_lowercase) for x in range(10))
-            target_fp = '%s.%s' % (path.join(self.tmp_dp, n), image_request.format)
+        n = ''.join(random.choice(string.ascii_lowercase) for x in range(10))
+        temp_fp = '%s.%s' % (path.join(self.tmp_dp, n), image_request.format)
```

Comment:
I tried doing this:
```
-        temp_fp = '%s.%s' % (path.join(self.tmp_dp, n), image_request.format)
+        temp_file = NamedTemporaryFile(dir=self.tmp_dp, suffix='.%s' % image_request.format)
+        temp_fp = temp_file.name
```
but several problems came up:
- the file-like object may go out of scope making `temp_fp` the path to a delete file, because it is automatically deleted
- it is not guaranteed that the path for the temporary file can be reused (`Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms`)
- in the caching use case, we need to atomically rename this temporary file into the cached one, it's not cleary how this plays with the deletion-on-garbage-collection
So I would avoid opening this rabbit hole